File: apps/android/test/se/scouterna/campfire/ConfigTests.kt
package se.scouterna.campfire

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

@DisplayName("the configuration")
class ConfigTests {
  @Test
  fun `reads the web origin the build settings put in BuildConfig`() {
    // A break anywhere between the product flavor, the generated BuildConfig, and the
    // source leaves the shell nowhere to load from. The tests always run the local
    // flavor, whose origin is this machine.
    assertEquals("http://localhost:8000", Config.webOrigin)
  }
}

File: apps/apple/test/ConfigTests.swift
import Foundation
import Testing
@testable import Campfire

@Suite("the configuration")
struct ConfigTests {
  @Test
  func `reads the web origin the build settings put in the Info.plist`() {
    // A break anywhere between the xcconfig, the generated plist, and the source leaves
    // the shell nowhere to load from. The tests always run the Local scheme, whose origin
    // is this machine.
    #expect(Config.webOrigin.scheme == "http")
    #expect(Config.webOrigin.host() == "localhost")
  }
}
